WIP Add KEP for Array Job in provisional status

alculquicondor opened the pull request:
Placeholder for early feedback

Commit:
Promote RunAsGroup to GA

irozzo-1A committed:
Extending Apiserver Network Proxy to handle traffic originated from Node network

andrewsykim committed:
KEP-1864: add graduation criterias

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-1864: add beta prod readiness review

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

Commit:
Updated Scaling Behavior KEP to reflect implementation

Signed-off-by: Arjun Naik <anaik@redhat.com>

PurelyApplied committed:
Update KEP-1933 owners, release state, and history.

* Ownership transfer from SIG-Instrumentation -> SIG-Security
* Current state is beta.
* Bring forward target for stable version in YAML - previous discussion indicated that beta state should be short-lived.
* Add Implementation History entries.

wojtek-t committed:
Strenghten KEP validation

adtac commented on keps/sig-apps/2214-array-job/kep.yaml:
```diff
@@ -0,0 +1,42 @@
+milestone:
+  alpha: "v1.21"
+  beta: "v1.22"
+  stable: "v1.23"
```
I think the graduation plan requires being in beta for two releases, so stable will be 1.24 at the earliest.

jayunit100 committed:
apiserver default labels ~ Implementable

- set to implementable and set versions progression
- add metric name apiserver_request_total
- add feature gate name
- remove feature gate in 123
- rephrase the breaking namespace description
- add apiserver rejection metric
- add failure mode around user labels

toc

example of apiserver metrics to monitor

justaugustus committed:
api: Make "title" and "owning-sig" fields required in KEPs

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
api: Make "kep-number" and "approver" fields required for PRR approvals

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/kepval: Move directory validation functionality to package

Signed-off-by: Stephen Augustus <foo@auggie.dev>

smarterclayton committed:
Update DaemonSet KEP for Alpha in 1.21

APIs and gate went into 1.20, impl will go into 1.21. Add some
guidance for new surge daemonset authors.

smarterclayton committed:
Update pod resource metrics KEP for beta with PRR

Pod resource metrics will go to beta in 1.21

umohnani8 committed:
Add KEP for enabling CRIContainerLogRotation by default

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

justaugustus committed:
kepval: Begin coalescing non-blocking warnings in test

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
api: Make "status" field required in KEPs

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
api: Move group helpers closer to the validation logic

Signed-off-by: Stephen Augustus <foo@auggie.dev>

jiahuif committed:
update text to reflect ongoing discussions.

jiahuif committed:
update KEP metadata.

justaugustus committed:
Ensure PRR approver is actually a valid approver

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
Revert "Merge pull request #2425 from justaugustus/refactor-revert"

This reverts commit 3e2c87bb7ec87125410631bce9dbd56373ce4aef, reversing
changes made to 0a495c8b528a970f136ed53a4310776b7fd17881.

Jiawei0227 committed:
Update CSI Migration Kep to include PRRQ

jiahuif committed:
add test plan.

mattcary committed:
Initial draft of KEP for alpha release

andrewsykim committed:
KEP-2086: prod readiness review

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-1959: clarify field naming to loadBalancerClass

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-1959: add note about using finalizer service.kubernetes.io/load-balancer-cleanup

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-1959: prod readiness review

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-2395: introduce feature gates DisableCloudProviders and DisableKubeletCloudCredentialProviders

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-2395: add prod readiness review for alpha

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-2395: update table of contents

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-2395: update kep.yaml based on new template

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

justaugustus committed:
kepval: Retrieve absolute paths for KEP and PRR directories

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
kepval: Use a map to coalesce errors based on filenames

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
kepval: Fix PRR approver search

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
kepval: Strip `@` from PRR approver during validation

Signed-off-by: Stephen Augustus <foo@auggie.dev>

Shekharrajak committed:
Migrating from old kep to new template: sig-network  keps

Migrating from old kep to new template: sig-network - updated the status

Migrating from old kep to new template: sig-network - NodeLocal DNS Cache is not migrated

added missing field in kep.yaml for 2449

justaugustus committed:
kepval: Temporarily skip PRR approval validation if "stage" is missing

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/legacy: Restore helpers.go

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
hack/verify-kep-metadata: Make `go test` verbose

Signed-off-by: Stephen Augustus <foo@auggie.dev>

andrewsykim committed:
KEP-2133: add beta milestone and prod readiness review

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-2133: add kubelet metrics for credential provider plugins

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

dashpole committed:
update tracing PRR to newer format

ahg-g committed:
Added KEP for scheduler node resource score strategy.

amwat committed:
add kep 2464 kubetest2 ci migration

bobbypage committed:
Update graceful shutdown KEP for beta

* Target beta in 1.21
* Add PRR review

amwat committed:
mark checklist items

bridgetkromhout committed:
Merge remote-tracking branch 'lachie83/563-dual-stack-kep-update' into updates-to-563

bridgetkromhout committed:
Typo fix

Signed-off-by: Bridget Kromhout <bridget@kromhout.org>

PurelyApplied committed:
[KEP-1933] Production Readiness Review.

ehashman committed:
Fill out PRR, graduation criteria, and implementation history

andrewsykim committed:
KEP-2086: update to include Topology traffic policy

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
KEP-2133: add beta test plan

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

k8s-ci-robot committed:
Merge pull request #2083 from jiahuif/kep-cloud-controller-manager-migration-lease-api

cloud controller manager migration: default values and lease types update.

mrunalp committed:
Update CRI kep

Update the kep with latest status and the prr file

Signed-off-by: Mrunal Patel <mpatel@redhat.com>

bswartz committed:
Include populator implementation details

k8s-ci-robot committed:
Merge pull request #2441 from andrewsykim/prr-2086

KEP-2086: alpha prod readiness review

robscott committed:
Adding KEP 2433: Topology Aware Hints (#2434)

k8s-ci-robot committed:
Merge pull request #2460 from deads2k/refactor

Migrating from old kep to new template - kep-number: 2330

Commit:
Correct yaml syntax

k8s-ci-robot committed:
Merge pull request #2440 from mattcary/statefulset-deletion

Autodelete StatefulSet PVCs

k8s-ci-robot committed:
Merge pull request #2466 from lauralorenz/fix-link-for-kep-0001

Fix links to the KEP on the KEP process

k8s-ci-robot committed:
Merge pull request #2286 from xing-yang/volumehealth_beta

Update volume health KEP

k8s-ci-robot committed:
Merge pull request #2428 from dashpole/tracing_121

Move tracing to 1.21 milestone

deads2k committed:
confirm PRR reviewer for insecure-skip-backend-proxy

k8s-ci-robot committed:
Merge pull request #2476 from deads2k/insecure-backup-prr

confirm PRR reviewer for insecure-skip-backend-proxy

k8s-ci-robot committed:
Merge pull request #2456 from bg-chun/memory-manager-proposal

Submitting a KEP for production readiness approval

k8s-ci-robot committed:
Merge pull request #2405 from Jefftree/ssa

Update Server Side Apply KEP for 1.21 GA

jiahuif committed:
rename due to KEP number change.

k8s-ci-robot committed:
Merge pull request #2457 from andrewsykim/kep-2133

KEP-2133: add beta milestone and prod readiness review

k8s-ci-robot committed:
Merge pull request #2418 from umohnani8/log

Add KEP for enabling CRIContainerLogRotation by default

k8s-ci-robot committed:
Merge pull request #2442 from andrewsykim/prr-1959

KEP-1959: add prod readiness review for alpha

k8s-ci-robot committed:
Merge pull request #1690 from verb/ec2

Propose removing and security context for Ephemeral Containers

thockin committed:
KEP 2200 (block externalIPs) is implemented

k8s-ci-robot committed:
Merge pull request #2244 from verb/ec-1.21-beta

Update Ephemeral Containers KEP for beta in 1.21

k8s-ci-robot committed:
Merge pull request #2416 from smarterclayton/ds_kep

Update DaemonSet KEP for Alpha in 1.21

andrewsykim committed:
KEP-1864: add feature gate to kep.yaml

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

bridgetkromhout committed:
Updates to PRR for KEP

Signed-off-by: Bridget Kromhout <bridget@kromhout.org>

k8s-ci-robot committed:
Merge pull request #2471 from ehashman/34-sysctls

Migrate KEP-34 to new template

BenTheElder committed:
add 2420 PRR request

amwat committed:
Update keps/sig-testing/2464-kubetest2-ci-migration/README.md

Co-authored-by: Sascha Grunert <sgrunert@suse.com>

bridgetkromhout committed:
Added to PRR

Signed-off-by: Bridget Kromhout <bridget@kromhout.org>

k8s-ci-robot committed:
Merge pull request #2473 from PurelyApplied/kep-1933-prr

[KEP-1933] Production Readiness Review.

lachie83 committed:
Merge pull request #3 from bridgetkromhout/updates-to-563

Additional updates to PRR docs

k8s-ci-robot committed:
Merge pull request #2443 from andrewsykim/cloud-provider-feature-gates

KEP-2395 Introduce feature gates DisableCloudProviders and DisableKubeletCloudCredentialProvider

k8s-ci-robot committed:
Merge pull request #2432 from Jiawei0227/master

Update CSI Migration Kep to include PRRQ

k8s-ci-robot committed:
Merge pull request #2066 from andrewsykim/kep-1864

KEP-1864: add prod readiness review and beta milestone

BenTheElder committed:
rename kep.yml kep.yaml

bridgetkromhout committed:
Additional edits for PRR

Signed-off-by: Bridget Kromhout <bridget@kromhout.org>

k8s-ci-robot committed:
Merge pull request #2472 from bobbypage/graceful-shutdown-beta-kep

KEP 2000: Update graceful shutdown KEP for beta

lachie83 committed:
Merge pull request #4 from bridgetkromhout/updates-to-563

Additional edits for PRR

k8s-ci-robot committed:
Merge pull request #2327 from lachie83/563-dual-stack-kep-update

Update dual stack KEP metadata to include status and PRR fields

BenTheElder committed:
correct PRR stage in PRR yaml

k8s-ci-robot committed:
Merge pull request #2469 from BenTheElder/implementable-build

Update Reducing Kubernetes Build Maintenance to implementable

k8s-ci-robot committed:
Merge pull request #2461 from ahg-g/ahg-fit-strategy

Added KEP for Node Resource Score Strategy

k8s-ci-robot committed:
Merge pull request #2284 from jayunit100/2161-impl

apiserver default labels ~ Implementable

jiahuif committed:
update KEP metadata to match new template.

jiahuif committed:
update KEP for PRR.

k8s-ci-robot committed:
Merge pull request #2475 from mrunalp/update_cri_kep

Update CRI kep with latest status

jiahuif committed:
use a feature gate instead of a sole flag.

amwat committed:
remove n/a parts

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

amwat committed:
add more clarity around graduation criteria

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

amwat committed:
fix kep.yaml

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

k8s-ci-robot committed:
Merge pull request #2417 from smarterclayton/resource_kep

Update pod resource metrics KEP for beta with PRR

amwat committed:
Add proper title

amwat committed:
hack/update-toc.sh

k8s-ci-robot committed:
Merge pull request #2056 from bswartz/volume-populators-beta

Reformat populators KEP

k8s-ci-robot committed:
Merge pull request #2465 from amwat/kubetest2

Add KEP 2464: Kubetest2 CI migration

YoyinZyc committed:
add feature gate to metric cardinality kep.

k8s-ci-robot committed:
Merge pull request #2477 from jiahuif/leader-migration-to-alpha

KEP-2436 PRR: Controller Manager Leader Migration to alpha

amwat committed:
Move Kubetest2 CI migration to implementable

amwat committed:
fix kep number

k8s-ci-robot committed:
Merge pull request #2025 from irozzo-1A/extend-konnectivity

KEP: Extending Apiserver Network Proxy to handle traffic originated from Node network

k8s-ci-robot committed:
Merge pull request #2130 from dashpole/remove_cadvisor_json

Add KEP for removing cAdvisor json metrics

k8s-ci-robot committed:
Merge pull request #2468 from YoyinZyc/update_kep

Update the metric cardinality kep

k8s-ci-robot committed:
Merge pull request #2479 from amwat/kubetest2-prr

Move Kubetest2 CI migration to implementable

k8s-ci-robot committed:
Merge pull request #1974 from krmayankk/runasgroup

Promote RunAsGroup to GA

justaugustus committed:
api: Add interface directives for counterfeiter

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
generated: Run make update-mocks

Signed-off-by: Stephen Augustus <foo@auggie.dev>

SergeyKanzhelev committed:
updated KEP and postponed  graduation date for one release

justaugustus committed:
api: Add error handling and test for retrieving a stage's PRR approver

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
keps: Fix incorrect stages on PRR targeting v1.21

Signed-off-by: Stephen Augustus <foo@auggie.dev>

k8s-ci-robot committed:
Merge pull request #2482 from justaugustus/content-fixes

keps: Fix incorrect stages on PRR targeting v1.21

ahg-g committed:
Remove the restriction on empty namespaceSelector in podaffinity

k8s-ci-robot committed:
Merge pull request #2483 from ahg-g/ahg-fit-strategy

Remove the restriction on empty namespaceSelector in podaffinity

k8s-ci-robot committed:
Merge pull request #2409 from justaugustus/validate

api: Update field validations for `Document`s

k8s-ci-robot committed:
Merge pull request #2452 from shekhar-rajak/kep_new_template_sig_storage

Migrating from old kep to new template: sig-storage

ehashman committed:
Add ehashman to PRR team

jpbetz committed:
Fix image links in KEP 2436

wojtek-t committed:
Merge two separate NodeLocalDNS KEPs

k8s-ci-robot committed:
Merge pull request #2486 from jpbetz/patch-1

Fix image links in KEP 2436

k8s-ci-robot committed:
Merge pull request #2481 from SergeyKanzhelev/dockershimRemovalPostponement

updated Dockershim KEP and postponed  graduation date for one release

SergeyKanzhelev committed:
Small typo on year

I meant 2022, not 2021

k8s-ci-robot committed:
Merge pull request #2446 from shekhar-rajak/kep_new_template_sig_network

Migrating from old kep to new template: sig-network

justaugustus committed:
api: Add NewKEPHandler method

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg: Remove references to legacy packages

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
.githooks/pre-commit: Fix shebang and simplify verifies

Signed-off-by: Stephen Augustus <foo@auggie.dev>

Shekharrajak committed:
Migrating from old kep to new template: sig-cluster-lifecycle

Migrating from old kep to new template: sig-cluster-lifecycle

Shekharrajak committed:
Migrating from old kep to new template: sig-cluster-lifecycle - kubeadm

Migrating from old kep to new template: sig-cluster-lifecycle - kubeadm

Migrating from old kep to new template: sig-cluster-lifecycle - kubeadm

README file from kubeadm to parent folder

removed README file - not needed

xiaoxubeii committed:
Update cgroup v2 deps status.

navidshaikh committed:
Fix typos in structured logging KEP README

Signed-off-by: Navid Shaikh <shaikhnavid14@gmail.com>

k8s-ci-robot committed:
Merge pull request #2493 from shekhar-rajak/kep_new_template_sig_cluster_lifecycle

Migrating from old kep to new template: sig-cluster-lifecycle

wojtek-t committed:
Minor kepval changes to better validate dir-based templates

pohly committed:
generic ephemeral volumes: document metrics

k8s-ci-robot committed:
Merge pull request #2511 from navidshaikh/pr/fix-typo

Fix typos in structured logging KEP README

k8s-ci-robot committed:
Merge pull request #2484 from ehashman/ehashman-prr

Add ehashman to PRR team

kikisdeliveryservice committed:
update README to link 1.21 sheet and link to open issue

k8s-ci-robot committed:
Merge pull request #2478 from thockin/externalips-admission

KEP 2200 (block externalIPs) is implemented

k8s-ci-robot committed:
Merge pull request #2491 from SergeyKanzhelev/patch-4

Small typo in year

k8s-ci-robot committed:
Merge pull request #2487 from wojtek-t/merge_node_local_cache_keps

Merge two separate NodeLocalDNS KEPs

deads2k committed:
elaborate on the choice to make spec.signer required for CSRs

soltysh committed:
Update KEP 1020 status to implemented

k8s-ci-robot committed:
Merge pull request #2516 from soltysh/update_staging_status

Update KEP 1020 status to implemented

k8s-ci-robot committed:
Merge pull request #2515 from deads2k/clarify-spec-signer

elaborate on the decision to make spec.signer required for CSRs

k8s-ci-robot committed:
Merge pull request #2499 from shekhar-rajak/kep_new_template_sig_cluster_lifecycle_kubeadm

Migrating from old kep to new template: sig-cluster-lifecycle - kubeadm

marosset committed:
KEP 1981: Windows privileged container KEP updates for alpha (#2288)

* Windows privileged container KEp alpha updates

* updagint TOC

* fixing lots of typos

* updates with feedback for liggitt

* fixing kep.yaml

* minor KEP feedback

* minor KEP feedback

* Rename fields to HostJob

* KEP.yaml updates

* Adding PRR yaml file and approver

* Fixing KEP metadata

* s/hostJob/hostProcess

* responding to feedback from jsturtevant

* fixing status in kep.yaml

* a few minor updates

* Clarifying container mounts section and validation rules section

* Updates from Liggitt's feedback

* fixup

* updating volume support

* updating info about how volume mount paths are derived

* minor updates

* Clarifying HostPID/HostIPC usage

* reworking hostPID section

wojtek-t committed:
Cleanup metadata in metrics stability kep

k8s-ci-robot committed:
Merge pull request #2521 from wojtek-t/canonical_files

Minor kepval changes to better validate dir-based templates

k8s-ci-robot committed:
Merge pull request #2520 from wojtek-t/update_metrics_stability

Cleanup metadata in metrics stability kep

k8s-ci-robot committed:
Merge pull request #2514 from kikisdeliveryservice/update-readme

update README to link 1.21 sheet and link issue template

wojtek-t committed:
Migrate remaining SIG api-machinery KEP to new template

ehashman committed:
Add log volume benchmark

k8s-ci-robot committed:
Merge pull request #2488 from ehashman/logging-test

Add log volume benchmark

dashpole committed:
add manual test for removal

dashpole committed:
remove Service from the apiserver configuration

dashpole committed:
rename Master egresstype to ControlPlane

dashpole committed:
rename OpenTelemetryClientConfiguration to TracingConfiguration

dashpole committed:
add sampling rate to configuration

dashpole committed:
clarify that traces are never collected when disabled

dashpole committed:
add user stories section

justaugustus committed:
Cleanup lint warnings

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg: Move proposal logic into separate package

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg: Copy kepctl logic into repo package

Signed-off-by: Stephen Augustus <foo@auggie.dev>

k8s-ci-robot committed:
Merge pull request #2080 from jayunit100/patch-4

keps/NNNN-kep-template: Add note about test refactors

justaugustus committed:
pkg: Update references to `repo` functions/methods

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
cmd: Update references to `repo` functions/methods

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
go.mod: Update github dependencies

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Parameterize commonly-used constants

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Move more configuration options into `Repo` struct

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Rely on `Repo` struct to retrieve paths

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Move repo validation into package

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Init `Document` handlers in `Repo` struct

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Fixup query tests

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg: Move kepctl testdata to repo package

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Copy latest templates into testdata

Signed-off-by: Stephen Augustus <foo@auggie.dev>

wojtek-t committed:
Migrate remaining cloud-provider KEPs to new template

justaugustus committed:
pkg/repo: Use handlers from the `Repo` instance when loading `Document`s

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Copy repo validation test in

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
kepctl: Move command logic into pkg/

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
kepctl: Update references to refactored packages

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Debugging for query logic

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Make PRR validation errors non-blocking when querying

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Initial cleanup on KEP creation logic

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
pkg/repo: Simplify proposal validation test

Signed-off-by: Stephen Augustus <foo@auggie.dev>

k8s-ci-robot committed:
Merge pull request #2524 from wojtek-t/migrate_apimachinery_keps

Migrate remaining SIG api-machinery KEP to new template

wojtek-t committed:
Remove unuseful README files

wojtek-t committed:
Move documentation-for-images to the right directory

k8s-ci-robot committed:
Merge pull request #2533 from wojtek-t/migrate_cloud_provider_keps

Migrate remaining cloud-provider KEPs to new template

serathius committed:
Add e2e tests requirements for Beta structured logging (#2529)

* Add e2e tests requirements for Beta structured logging

* Update keps/sig-instrumentation/1602-structured-logging/README.md

Co-authored-by: Marek Siarkowicz <serathius@users.noreply.github.com>

k8s-ci-robot committed:
Merge pull request #2536 from wojtek-t/strenghten_kep_validation

Enforce KEP number and fixup extraneous markdown documents

k8s-ci-robot committed:
Merge pull request #2525 from dashpole/cadvisor_test_plan

Add manual test for removal of endpoints

wojtek-t committed:
Update KEP 1904 with upgrade test results

chaodaiG committed:
Add a KEP at sig-testing: continuously deploy k8s prow

k8s-ci-robot committed:
Merge pull request #2538 from wojtek-t/update_efficient_watch_resumption_kep

Update KEP 1904 with upgrade test results

justaugustus committed:
pkg/repo: Remove logic for processing legacy KEPs

Signed-off-by: Stephen Augustus <foo@auggie.dev>

alculquicondor committed:
Update target stable milestone for scheduler CC

We are releasing a v1beta2 in 1.21 that we intend to graduate to v1 next release.

With it, scheduling profiles will graduate too.

Signed-off-by: Aldo Culquicondor <acondor@google.com>

k8s-ci-robot committed:
Merge pull request #2429 from ehashman/move-log-redaction

Move log sanitization to SIG Security

k8s-ci-robot committed:
Merge pull request #2542 from alculquicondor/patch-3

Update target stable milestone for scheduler CC

k8s-ci-robot committed:
Merge pull request #2528 from justaugustus/cleanup

Repo package cleanups/refactors (part two)

k8s-ci-robot committed:
Merge pull request #2303 from PurelyApplied/kep-1933-update

Update KEP-1933 owners, release state, and history.

MikeSpreitzer committed:
Switch from serial to concurrent transmision in virtual world

Also fix note on complexity of maintaining mu(i,t).

chaodaiG committed:
Annotated prow job with plank version https://github.com/kubernetes/test-infra/pull/21054

umohnani8 committed:
Update CRIContainerLogRotation KEP

Update the graduation criteria for GA as one of the
issues mentioned no longer applies.

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

jiahuif committed:
fix errors in sample configuration and code indent.

neolit123 committed:
1602-structured-logging: add note about excluded components

A prior discussion determined kubectl and kubeadm
as components that should not be migrated.

pohly committed:
generic ephemeral volumes: document tests

justaugustus committed:
OWNERS: Update Enhancements subproject owners

- Add Kirsten as subproject owner
- Add Anna and Nabarun as enhancements-reviewers
- Add Nabarun to kep-tools-reviewers

Signed-off-by: Stephen Augustus <foo@auggie.dev>

k8s-ci-robot committed:
Merge pull request #2549 from neolit123/1.21-update-instrumentation-kep-exclude-kubectl-kubeadm

1602-structured-logging: add note about excluded components

k8s-ci-robot committed:
Merge pull request #2540 from chaodaiG/kep-prow-cd

KEP: (sig-testing) continuously deploy k8s prow

chaodaiG committed:
KEP-2539: Addressing comments from #2540

k8s-ci-robot committed:
Merge pull request #2550 from justaugustus/moar-owners

OWNERS: Update Enhancements subproject owners

dashpole committed:
remove another reference to egress to service

dashpole committed:
requests ask for sampling

Co-authored-by: Daniel Smith <dbsmith@google.com>

k8s-ci-robot committed:
Merge pull request #2526 from dashpole/apiserver_tracing_updates

Apiserver tracing updates

k8s-ci-robot committed:
Merge pull request #2507 from xiaoxubeii/dep-status-updates

Update cgroup v2 deps status

Commit:
suspend jobs: add note about StartTime and ADS

Signed-off-by: Adhityaa Chandrasekar <adtac@google.com>

k8s-ci-robot committed:
Merge pull request #2557 from adtac/ads

suspend jobs: add note about StartTime and ADS

k8s-ci-robot committed:
Merge pull request #2547 from umohnani8/log

Update CRIContainerLogRotation KEP

k8s-ci-robot committed:
Merge pull request #2544 from MikeSpreitzer/concurrent-queues

Swich from serial to concurrent transmision in virtual world

alculquicondor committed:
Update implementation history for Indexed Job

Incorporate fixes done during implementation.

Signed-off-by: Aldo Culquicondor <acondor@google.com>

soltysh committed:
Replace prydonius with soltysh for sig-apps leads

k8s-ci-robot committed:
Merge pull request #2575 from soltysh/soltysh_apps

Replace prydonius with soltysh for sig-apps leads

k8s-ci-robot committed:
Merge pull request #2573 from alculquicondor/array-job

Update implementation history for Indexed Job

tallclair committed:
Delete blank line from .notableofcontents

k8s-ci-robot committed:
Merge pull request #2548 from jiahuif/kep-2436/config-fix

(KEP-2436) fix errors in sample configuration and formatting.

damemi committed:
Rename RandomReplicaSetDownScale feature gate to LogarithmicScaleDown

k8s-ci-robot committed:
Merge pull request #2584 from damemi/rename-random-downscale

Rename RandomReplicaSetDownScale feature gate to LogarithmicScaleDown

k8s-ci-robot committed:
Merge pull request #2553 from chaodaiG/prow-deploy-address-comment

KEP-2539: Addressing comments from #2540

thockin committed:
API modifications for in-place pod resize KEP (#1883)

Add a per-pod "resize" field and make a bunch of clarifications on the
overall flow of an in-place update.

Commit:
Small fixes

Signed-off-by: Arjun Naik <anaik@redhat.com>

k8s-ci-robot committed:
Merge pull request #2159 from arjunrn/update-scaling-behavior-kep

Updated Scaling Behavior KEP to reflect implementation

k8s-ci-robot committed:
Merge pull request #2513 from pohly/generic-inline-volumes

generic ephemeral volumes: document metrics and tests

k8s-ci-robot committed:
Merge pull request #2583 from tallclair/toc

Delete blank line from .notableofcontents

k8s-ci-robot committed:
Merge pull request #2375 from natasha41575/OpenApiFeaturesInKustomize

change openapi features in kustomize KEP status to implementable

Commit:
update status on metric stability framework

Commit:
update implementation history and link relevant PRs

Commit:
fix toc

k8s-ci-robot committed:
Merge pull request #2612 from logicalhan/stability-ga

update status on metric stability framework

wojtek-t committed:
Mark Immutable secrets/configmaps KEP as implemented

alculquicondor committed:
Graduate Indexed Job to beta

Add metrics and SLOs

Signed-off-by: Aldo Culquicondor <acondor@google.com>
